host/CMakeLists.txt

```diff
@@ -1,23 +1,68 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License.
 
+# Function to set the proper compile definitions for the given target.
+function(set_host_compile_defs TARGET FLAGS)
```

Comment:
Probably an easier / more CMake-y way to go about this is to create an `INTERFACE` host library with all these settings (except the `install`) and then just `add_library(foo) target_link_libraries(host-app) install(foo)`.